doc/message_dispatching.puml

```diff
@@ -0,0 +1,52 @@
+
+@startuml
+
+interface MessageDispatcherInterface {
+    +dispatch_call(const json& call, bool triggered = false)
+    +dispatch_call_async(const json& call, bool triggered = false): std::future<EnhancedMessage<T>>
+    +dispatch_call_result(const json& call_result)
+    +dispatch_call_error(const json& call_error)
+    +new_message_id(): MessageId
```

Comment:
Do we even need to bother with message IDs for users of this layer? The layer should be able to take care of it itself right?

Reply:
Yes, this functionality does not necessarily have to be part of this interface

Reply:
Looking at it, I think we might want to move the `createMessageId` functionality out of the message queue, it can be a free function. Then on construction of a `Call<T>` type we can generate a message id internally and essentially hide this value.